client/order/index.js

```diff
@@ -42,30 +44,53 @@ jQuery( function ( $ ) {
 			'wc-refunded' === this.value &&
 			'wc-refunded' !== originalStatus
 		) {
-			if ( ! canRefund ) {
-				alert(
-					__( 'Order cannot be refunded', 'woocommerce-payments' )
-				);
-				return;
-			}
-			if ( 0 >= refundAmount ) {
-				alert( __( 'Invalid Refund Amount', 'woocommerce-payments' ) );
+			renderRefundConfirmationModal(
+				originalStatus,
+				canRefund,
+				refundAmount
+			);
+		} else if (
+			'wc-cancelled' === this.value &&
+			'wc-cancelled' !== originalStatus
+		) {
+			if ( ! canRefund || 0 >= refundAmount ) {
 				return;
 			}
-			const container = document.createElement( 'div' );
-			container.id = 'wcpay-refund-confirm-container';
-			document.body.appendChild( container );
-			ReactDOM.render(
-				<RefundConfirmationModal
-					orderStatus={ originalStatus }
-					refundAmount={ refundAmount }
-					formattedRefundAmount={ getConfig(
-						'formattedRefundAmount'
-					) }
-					refundedAmount={ getConfig( 'refundedAmount' ) }
-				/>,
-				container
+			renderModal(
+				<CancelConfirmationModal
+					originalOrderStatus={ originalStatus }
+				/>
 			);
 		}
 	} );
+
+	function renderRefundConfirmationModal(
+		originalStatus,
+		canRefund,
+		refundAmount
+	) {
+		if ( ! canRefund ) {
+			alert( __( 'Order cannot be refunded', 'woocommerce-payments' ) );
```

Comment:
Just a suggestion: is it possible to use an internal library dialog UI component instead of an alert? 
IMO, alerts are harsh and outdated.

Reply:
Will check, was going by the fact that existing refund errors use an alert here on order edit
https://github.com/Automattic/woocommerce/blob/5ed070d78a840fa18e7e89384fc8bdf3e827ee6e/plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js#L998